code/game/gamemodes/wizard/spellbook.dm

```diff
@@ -24,383 +74,214 @@
 			qdel (O)
 			O = null
 
+#define buy_href_link(obj, price, txt) ((price > uses) ? "Price: [price] point\s" : "<a href='?src=\ref[src];spell=[obj];buy=1'>[txt]</a>")
+
 /obj/item/weapon/spellbook/attack_self(mob/user = usr)
 	if(!user)
 		return
+
 	user.set_machine(src)
+
 	var/dat
-	if(temp)
-		dat = "[temp]<BR><BR><A href='byond://?src=\ref[src];temp=1'>Clear</A>"
+	dat += "<head><title>Spellbook ([uses] REMAINING)</title></head><body>"
+	dat += "<h1>Edvin's Catalogue Of Spells And Artifacts</h1><br>"
+	dat += "<h2>[uses] points remaining (<a href='?src=\ref[src];refund=1'>Get a refund</a>)</h2><br>"
+	dat += "<i>Penned by Edvin The Starcrusher.</i><br>"
+	dat += "<i>This book contains a list of many useful things that you'll need in your journey.</i><br>"
+	dat += "<b>SPELLS:</b><br><br>"
+
+	for(var/spell_path in available_spells)
+		var/spell/abstract_spell = spell_path
+		var/spell/spell = locate(spell_path) in user.spell_list
+
+		if(istype(spell)) //User knows the spell
+			//FORMATTING
+
+			//<b>Fireball</b> - 10 seconds<br>
+			//Requires robes to cast
+			//speed: 1/5 (upgrade) | power: 0/1 (upgrade)
+
+			var/spell_name = spell.name
+			var/spell_cooldown = get_spell_cooldown_string(spell.charge_max, spell.charge_type)
+
+			dat += "<b>[spell_name]</b> - [spell_cooldown]<br>"
+
+			//Get spell properties
+			var/list/properties = get_spell_properties(spell.spell_flags, user)
+			var/property_data
+			for(var/P in properties)
+				property_data += "[P] "
+
+			if(property_data)
+				dat += "[property_data]<br>"
+
+			//Get the upgrades
+			var/upgrade_data = ""
+			for(var/upgrade in spell.spell_levels)
+				var/lvl = spell.spell_levels[upgrade]
+				var/max = spell.level_max[upgrade]
+
+				//If maximum upgrade level is 0, skip
+				if(0 == max)
+					continue
+
+				upgrade_data += "<a href='?src=\ref[src];spell=\ref[spell];upgrade_type=[upgrade];upgrade_info=1'>[upgrade]</a>: [lvl]/[max] (<a href='?src=\ref[src];spell=\ref[spell];upgrade_type=[upgrade];upgrade=1'>upgrade</a>)  "
+
+			if(upgrade_data)
+				dat += "[upgrade_data]<br><br>"
+		else //User doesn't know the spell
+			//FORMATTING
+
+			//<b>Fireball</b> - 10 seconds (buy for 1 spell point)
+			//<i>(Description)</i>
+			//Requires robes to cast
+
+			var/spell_name = initial(abstract_spell.name)
+			var/spell_cooldown = get_spell_cooldown_string(initial(abstract_spell.charge_max), initial(abstract_spell.charge_type))
+			var/spell_price = get_spell_price(abstract_spell)
+
+			dat += "<b>[spell_name]</b> - [spell_cooldown] ([buy_href_link(spell_path, spell_price, "buy for [spell_price] point\s")])<br>"
```

Comment:
<b> is deprecated, and <i> is no longer directly italics in HTML5: instead, you should use <strong> and <em> respectively.